includes/integrations/wc-blocks/class-orddd-lite-delivery-blocks.php

```diff
@@ -0,0 +1,337 @@
+					if ( '' !== $total_fees && null !== $total_fees ) {
+						Orddd_Lite_Common::update_order_meta( $order_id, '_total_delivery_charges', $total_fees, $order );
+						WC()->session->__unset( '_total_delivery_charges' );
+					} else {
+						Orddd_Lite_Common::update_order_meta( $order_id, '_total_delivery_charges', '0', $order );
+					}
+
+					self::orddd_lite_update_lockout_timeslot( $h_deliverydate, $order_time_slot );
+				}
+				$order->save();
+			}
+
+			do_action( 'orddd_after_timeslot_update', $time_slot, $order_id );
+		}
+	}
+	/**
+	 * Update number of order for Delivery date and Time slot in options table
+	 *
+	 * @globals resource $wpdb WordPress Object
+	 *
+	 * @param string $delivery_date Selected Delivery date on the checkout page.
+	 * @param string $timeslot Selected time slot on the checkout page.
+	 *
+	 * @since 3.11.0
+	 */
+	public static function orddd_lite_update_lockout_timeslot( $delivery_date, $timeslot ) {
+		if ( '' === $timeslot || __( 'As Soon As Possible.', 'order-delivery-date' ) === $timeslot ) {
+			return;
+		}
+
+		$lockout_date        = $delivery_date;
+		$time_format_to_show = Orddd_Lite_Common::orddd_lite_get_time_format();
+		$lockout_time        = get_option( 'orddd_lite_lockout_time_slot' );
+		if ( '' == $lockout_time || '{}' == $lockout_time || '[]' == $lockout_time || 'null' == $lockout_time ) { //phpcs:ignore
+			$lockout_time_arr = array();
+		} else {
+			$lockout_time_arr = json_decode( $lockout_time );
+		}
+		$existing_timeslots   = array();
+		$existing_dates       = array();
+		$lockout_time_new_arr = array();
+		$total_quantities     = 1;
+		$timeslot             = Orddd_Lite_Common::orddd_lite_change_time_slot_format( $timeslot, $time_format_to_show );
+
+		foreach ( $lockout_time_arr as $k => $v ) {
+			$orders = $v->o;
+			if ( $timeslot == $v->t && $lockout_date == $v->d ) { //phpcs:ignore
+				$orders = $v->o + $total_quantities;
+			}
+			$existing_timeslots[ $v->d ][] = $v->t;
+			$existing_dates[]              = $v->d;
+			$lockout_time_new_arr[]        = array(
+				'o' => $orders,
+				't' => $v->t,
+				'd' => $v->d,
+			);
+		}
+
+		// add the currently selected date if it does not already exist.
+		if ( ( isset( $existing_timeslots[ $lockout_date ] ) && ! in_array( $timeslot, $existing_timeslots[ $lockout_date ], true ) ) || ! in_array( $lockout_date, $existing_dates, true ) ) {
+			$lockout_time_new_arr[] = array(
+				'o' => $total_quantities,
+				't' => $timeslot,
+				'd' => $lockout_date,
+			);
+		}
+
+		$lockout_time_jarr = wp_json_encode( $lockout_time_new_arr );
+		update_option( 'orddd_lite_lockout_time_slot', $lockout_time_jarr );
+	}
+
+	/**
+	 * Validate delivery date field
+	 *
+	 * @hook woocommerce_checkout_process
+	 * @param object $order Order.
+	 * @param array  $request Order.
+	 * @since 1.4
+	 **/
+	public static function orddd_lite_validate_date_wpefield_cart_block( $order, $request ) {
```

Comment:
Call this function inside orddd_lite_update_block_order_meta_delivery_date itself as the hook for both is same. Call this before saving data